From a34717465da31b1d5519e09b3e32202380c0a6ab Mon Sep 17 00:00:00 2001
From: Brian Harring <ferringb@gmail.com>
Date: Sat, 19 May 2012 17:28:54 -0700
Subject: [PATCH] ebuild envs assert: ensure a zero exit code.

_unpacker within unpacker eclass is sensitive to it, thus
match portage behaviour.
---
 NEWS                                       |    4 ++++
 pkgcore/ebuild/eapi-bash/exit-handling.lib |    2 +-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/NEWS b/NEWS
index 268b79a..dace98a 100644
--- a/NEWS
+++ b/NEWS
@@ -5,6 +5,10 @@ See ChangeLog for full commit logs; this is summarized/major changes.
 
 pkgcore trunk:
 
+* Gentoo's unpacker eclass is sensitive to the return code of
+  assert; this is outside of pms rules, but we've matched portage
+  behaviour to keep things working
+
 * Fixed pinspect portageq envvar support.
 
 * Added `pconfig world` for world file manipulation.
diff --git a/pkgcore/ebuild/eapi-bash/exit-handling.lib b/pkgcore/ebuild/eapi-bash/exit-handling.lib
index bf2fcdb..c02b2db 100644
--- a/pkgcore/ebuild/eapi-bash/exit-handling.lib
+++ b/pkgcore/ebuild/eapi-bash/exit-handling.lib
@@ -5,7 +5,7 @@ assert() {
 	local _pipestatus=${PIPESTATUS[*]}
 	local x
 	for x in ${_pipestatus}; do
-		[ "$x" != 0 ] && die "$@"
+		[ "$x" -eq 0 ] || die "$@"
 	done
 }
 
-- 
1.7.8.5

